src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1804,6 +1938,38 @@ def nic_attach_choices(self):
         """
         return self.middleware.call_sync('interface.choices', {'exclude': ['epair', 'tap', 'vnet']})
 
+    @accepts()
+    def pptdev_choices(self):
+        """
+        Available choices for PCI passthru device.
+        """
+        if not self.pptdevs:
+            proc = None
+            try:
+                proc = subprocess.Popen(['/usr/sbin/pciconf', '-l'],
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.DEVNULL, text=True)
+                outs, errs = proc.communicate(timeout=5)
+            except (subprocess.TimeoutExpired, OSError, ValueError) as e:
+                if proc is not None:
+                    proc.kill()
+                    outs, errs = proc.communicate()
+                self.logger.warn(f'An error occured when checking the PCI configuration '
+                                 f'for devices available for PCI passthru. {e}.')
+            else:
+                if proc.returncode < 1:
```

Comment:
Is `-1` returncode okay ? I suspect okay would be `0`, can you please confirm ?

Reply:
Don't know of a situation where return code would be `<0` (except if terminated due to a signal). Unlikely, but not impossible this could happen, so agree it would be prudent to be conservative and only accept `0`. Better throw an extra error which can be analysed than risk intermittent malfunction with no clues why. Done.